packages/core/src/V.spec.ts

```diff
@@ -989,12 +995,75 @@ describe('number', () => {
         });
       }
     }
+
     test('should maintain original order even if execution order is reversed', async () => {
       const waitValidator = new WaitValidator();
       await expectValid([50, 40, 30, 20, 10, 1], V.array(waitValidator))
       expect(waitValidator.executionOrder).toEqual([1, 10, 20, 30, 40, 50])
     })
   })
+
+  describe('notEmpty', () => {
+    describe('valid cases', () => {
+      test('valid string input', async () => {
```

Comment:
These  (both valid and invalid cases separately) can be combined using `test.each`: https://vitest.dev/api/#test-each